heap_sort/HeapSort.java

```diff
@@ -0,0 +1,44 @@
+/*
+*  Following is the implementation of HeapSort algorithm
+*  It will sort the input array with time complexity O(N*logN)
+*  N being the size of array
+*/
+
+public class HeapSort {
+
+	public static void heapify(int[] arr, int size, int parent) {  // Build max-heap of subtree of parent index
```

Comment:
I've use `heapify` because I'm trying to create a more general sorting algorithm so that if we want to sort in descending we would need min heap which can be done from this one only by few changes.
Also CONTRIBUTING.md mentions only about the master to be same as file name but doesn't mention anything about naming conventions of supportive functions which I guess then would give us liberty to use what we wish. 😉